linetools/analysis/voigt.py

```diff
@@ -0,0 +1,203 @@
+            flg_rebin = 1
+        else:
+            wave = iwave
+    else: # No evaluation 
+        wave = iwave
+
+    # Line input
+    if isinstance(line,AbsLine):  # Single line as a AbsLine Class
+        pass
+        #par = [0*i for i in range(6)] # Dummy list
+        #par[0] = line.attrib['N'] # logN; Won't have units
+        #par[1] = line.attrib['z']
+        #par[2] = line.attrib['b'] # Should have units
+        #par[3] = line.wrest # Should have units
+        #par[4] = line.data['f']
+        #par[5] = line.data['gamma'] # Should have units
+    elif isinstance(line,list):
+        if isinstance(line[0],AbsLine):  # List of Abs_Line
+            tau = np.zeros(len(wave))
+            for iline in line:
+                if debug:
+                    print(iline)
+                tau += voigt_model(wave, iline, fwhm=0., 
+                    flg_ret=2, debug=debug, skip_wveval=True) 
+        else:
+            raise ValueError('voigt: Unknown type for voigt line in your list')
+    else: 
+        raise ValueError('voigt: Unknown type for voigt line')
+
+    # tau
+    if 'tau' not in locals(): # Deals with multiple line input
+        cold = 10.0**line.attrib['N'] / u.cm / u.cm
+        zp1=line.attrib['z']+1.0
+
+        wv=line.wrest.to(u.cm) #*1.0e-8
+        nujk = (const.c / wv).to(u.Hz)
+        dnu = (line.attrib['b'].to(u.km/u.s) / wv).to('Hz')
+        if line.data['gamma'].value == 0.:
+            warnings.warn('Gamma value is probably not set for wrest={:g} {}!'
+                .format(float(line.wrest.value),line.wrest.unit))
+        avoigt = (line.data['gamma']/( 4 * np.pi * dnu)).to(u.dimensionless_unscaled)
+
+        uvoigt = ( ((const.c / (wave/zp1)) - nujk) / dnu).to(u.dimensionless_unscaled)
+
+        # Voigt
+        cne = 0.014971475 * cold * line.data['f'] * u.cm * u.cm * u.Hz
+        try:
+            tau = cne * voigtking(uvoigt,avoigt) / dnu #(np.sqrt(np.pi) * dnu)
+        except IndexError:
+            pass
+        tau = tau.value # Should be dimensionless
+
+    # Only tau?
+    if flg_ret == 2:
+        if flg_rebin: # Can only occur for single line input
+            # Might be better to do in flux space, but small flux values cause trouble
+            tmodel = XSpectrum1D.from_tuple((wave,tau))
+            tmodel = tmodel.rebin(iwave)
+            tau = tmodel.flux.value
+            warnings.warn('Rebinned tau back to your input array.  Reconsider input')
+        # Return
+        return tau
+
+    # Flux
+    flux = np.exp(-1.0*tau)
+    vmodel = XSpectrum1D.from_tuple((wave,flux))
+
+    # Rebin?
+    if flg_rebin:
+        vmodel = vmodel.rebin(iwave)
+    # Convolve
+    if fwhm is not None:
+        vmodel.gauss_smooth(fwhm=fwhm)
+    else:
+        warnings.warn('Assuming infinite spectral resolution, i.e. no smoothing.')
+        warnings.warn('Set fwhm to smooth.')
+    
+    # Return
+    ret_val = []
+    if flg_ret % 2 == 1: ret_val.append(vmodel)
```

Comment:
Like I said above, I would prefer to use strings as flags instead of integers. I know that means we can't combine different flags to return multiple things (i.e. flg_ret=5 returns both the vmodel and flux).

Maybe allow multiple strings to return multiple values? i.e. ('tau', 'flux', 'vmodel')?  Then the order of the returned values would also be clear. Currently you have to work out the return order by looking at the code, or trial and error.